src/main/java/uk/nhs/digital/nhsconnect/nhais/service/RegistrationConsumerService.java

```diff
@@ -20,17 +23,27 @@
     @Autowired
     private InboundGpSystemService inboundGpSystemService;
 
+    @Autowired
+    private InboundStateRepository inboundStateRepository;
+
     public void handleRegistration(MeshMessage meshMessage) {
         LOGGER.debug("Received Registration message: {}", meshMessage);
         Interchange interchange = edifactParser.parse(meshMessage.getContent());
         LOGGER.debug("Parsed registration message into interchange: {}", interchange);
+
+        var inboundState = InboundState.fromInterchange(interchange);
+        try {
+            inboundStateRepository.save(inboundState); // this can detect duplicates as there is an unique compound index
+            LOGGER.debug("Saved inbound state: {}", inboundState);
+        } catch (DuplicateKeyException ex) {
```

Comment:
Shouldn't we handle this at call-site? Caller might decide what to do with duplicate (and tests would be more obvious)

Reply:
The idea is that for inbound messages (not recep) we need to skip the message if it's a duplicate. We will know if it's a duplicate when saving to DB (unique constraint violation). To be able to do that, we need to translate from mesh string message to edifact, build the inbound state record and save it.

Where would you like to put it?

Reply:
I thought about just not catching this exception here and leave it to caller.